system_metrics_collector/package.xml

```diff
@@ -16,14 +16,19 @@
   <build_depend>rcutils</build_depend>
   <build_depend>rcpputils</build_depend>
 
-  <!--Required for example launch file-->
-  <exec_depend>demo_nodes_cpp</exec_depend>
   <exec_depend>metrics_statistics_msgs</exec_depend>
   <exec_depend>rclcpp</exec_depend>
   <exec_depend>rclcpp_components</exec_depend>
   <exec_depend>rclcpp_lifecycle</exec_depend>
   <exec_depend>rcpputils</exec_depend>
   <exec_depend>rcutils</exec_depend>
+  <!--Required for e2e test file-->
+  <exec_depend>rclpy</exec_depend>
```

Comment:
Should it be test_depend for all these?

Reply:
Isn't test depend only for [unit tests](http://wiki.ros.org/catkin/package.xml#Dependencies)?

Reply:
I'm not sure we should be making a technical distinction between unit tests and integration tests - only a semantic one

Reply:
e.g. see https://github.com/ros2/launch/tree/master/launch_testing for one of the tools that is used to test launchfiles in packages. In ROS1, we used http://wiki.ros.org/rostest to launch multi-node test environments, but I think that's easier to just do now via regular channels